samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/README.md

```diff
@@ -4,14 +4,15 @@ This is a sample template for {{ cookiecutter.project_name }} - Below is a brief
 
 ```bash
 .
-├── README.md                   <-- This instructions file
+├── README.MD                   <-- This instructions file
+├── event.json                  <-- API Gateway Proxy Integration event payload
 ├── hello_world                 <-- Source code for a lambda function
-│   ├── app.js                  <-- Lambda function code
-│   ├── package.json            <-- NodeJS dependencies
-│   └── tests                   <-- Unit tests
-│       └── unit
-│           └── test_handler.js
-└── template.yaml               <-- SAM template
+│   └── app.js                  <-- Lambda function code
+├── package.json                <-- NodeJS dependencies and scripts
```

Comment:
package.json must be within the function folder. Each function should get its own manifest - that's the best practice we heard from customers and we are codifying that within "sam build" as the default (obviously overridable)

Reply:
Right! Should we also move ``tests/`` within the ``hello_world`` folder? If not, we'd also need to change `npm run test` behaviour.

Lemme know.

----

For context, under the Testing section we have:

```bash
npm install
npm run test
```

Reply:
The tests are already in the `hello_world` folder: https://github.com/awslabs/aws-sam-cli/tree/develop/samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/%7B%7Bcookiecutter.project_name%7D%7D/hello_world

Reply:
ha! That's what happens when you have multiple copies of it locally. Resuming work on this, thanks @jfuss for pointing this out !!!!

Reply:
Okay, I had to make another change so an user doesn't have to keep jumping in and out to run test, come back to where `template.yaml` is to run `sam local invoke` again, etc.

I included `npm run start` and `npm run invoke` scripts under `hello_world/package.json` for that matter.

Ideally, I **think**, for NodeJS, it'd be great to have a root `package.json` so one doesn't have to keep jumping to each source code folder and can do something like while still getting the benefit of having a per-function dependency:

* `npm run test:hello-world`: This would take care of `sam build && npm run test --prefix hello_world/`
* `npm run invoke:hello-world`:  This would take care of `sam local invoke HelloWorldFunction -e event.json`

Which then can be duplicated as one creates more functions and ease the project to grow organically. 

Source code tree as an example to visualize this:
```bash
.
├── README.md
├── event.json
├── hello_world
│   ├── app.js
│   ├── etc
│   ├── package.json                    <-- NodeJS dependencies and scripts
│   └── tests
│       └── unit
│           └── test_hello_world.js
├── package.json                        <-- NodeJS no deps w/ scripts for convenience
└── template.yaml
```

What are your thoughts? @jfuss @sanathkr

Reply:
@heitorlessa Any changes here are going to conflict with #845, where we are updating the nodejs examples to support `sam build` for Nodejs Lambdas. 

My concern with having two package.json is: How do I know which one to use or update?

Reply:
That's a totally valid concern @jfuss. After some thought and playing with `--prefix` I believe that will come down to preference at the end of the day. 

Customers I worked with on Node with multiple functions in a repo they had this approach, others used a Makefile so no standard per se. 

For this I will hold on any changes to this PR as we agree on what we ultimately prefer as a quickstart

Reply:
https://github.com/awslabs/aws-sam-cli/pull/845#issuecomment-446888460